packages/aws-cdk-lib/aws-codedeploy/lib/private/utils.ts

```diff
@@ -31,25 +31,56 @@ export function arnForDeploymentConfig(name: string, resource?: IResource): stri
   });
 }
 
-export function renderAlarmConfiguration(alarms: cloudwatch.IAlarm[], ignorePollAlarmFailure: boolean | undefined, removeAlarms = true):
-CfnDeploymentGroup.AlarmConfigurationProperty | undefined {
+export interface renderAlarmConfigProps {
+  /**
+   * Array of Cloudwatch alarms
+   */
+  readonly alarms: cloudwatch.IAlarm[],
+  /**
+   * Whether to ignore failure to fetch the status of alarms from CloudWatch
+   */
+  readonly ignorePollAlarmFailure?: boolean,
+  /**
+   * When no alarms are provided on an update, removes previously existing alarms from the construct.
```

Comment:
Please correct me if I'm wrong, but I believe this is what's happening. It was pretty unclear to me what exactly this did so I wanted to make sure it got documented.